### PR TITLE
HTCONDOR-868: Improve documentation for non-commercial S3 providers

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
Make a number of small changes to improve the S3 documentation.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
A preview of the new documentation is (temporarily) available at http://azaphrael.org/indexed/html/users-manual/file-transfer.html#transferring-files-using-the-s3-protocol.

File: docs/users-manual/file-transfer.rst [hunk 1]
```diff
@@ -718,43 +718,53 @@ and to upload ``output.txt`` to ``my_private_files/output.txt`` on a second acco
...
+for specific examples.
+
+HTCondor does not currently support transferring entire buckets or directories
+from S3.  However, if you specify an ``s3://`` URL as the
+``output_destination``, that URL will be used a prefix for each output file's
+location; if you specify a URL ending a ``/``, it will be treated like a
+directory.
+
+You may also use S3 URLs in ``transfer_output_remaps``.
+
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply docs/users-manual/file-transfer.rst:745]: Not at all.  Heading added.

File: docs/users-manual/file-transfer.rst [hunk 2]
```diff
@@ -718,43 +718,53 @@ and to upload ``output.txt`` to ``my_private_files/output.txt`` on a second acco
...
+storage servers using the S3 protocol via ``s3:// URLs``.  Downloading or
+uploading requires
 a two-part credential: the "access key ID" and the "secret key ID".  HTCondor
 does not transfer these credentials off the submit node; instead, it uses
 them to construct "pre-signed" ``https://`` URLs that temporarily allow
 the bearer access.  (Thus, an execute node needs to support ``https://``
 URLs for S3 URLs to work.)
 
-To make use of this feature, specify a file containing
+To make use of this feature, you'll need to specify a file containing
```
[the remark this answers is not in the document]
Comment (ChristinaLK) [reply docs/users-manual/file-transfer.rst]: Secondary note to this: I feel like this whole paragraph could be clearer, something like: 
```
To make use of this feature, you will need to specify the following information in the submit file: 
-  a file containing your secret access key (and nothing else)
- a file containing your access key ID (and nothing else)
- S3 URLs as either input values or output destinations
```

File: docs/users-manual/file-transfer.rst [hunk 3]
```diff
@@ -786,3 +796,19 @@ approach defined above. e.g.
...
+key files described above.
+
+.. code-block:: condor-submit
+
+    s3_access_key_id_file = /home/example/secrets/accessKeyID
+    s3_secret_access_key_file = /home/example/secrets/secretAccessKey
+    transfer_input_files = s3://some.other-s3-provider.org/my-bucket/large-input.file
+    transfer_output_remaps = "large-output.file = s3://some.other-s3-provider.org/my-bucket/large-output.file"
+
+    # If you need to specify a region, you can do so.
```
[the remark this answers is not in the document]
Comment (ChristinaLK) [reply docs/users-manual/file-transfer.rst:816]: I assumed this was copy-paste carryover. If it's not, I think a comment indicating that region can be relevant even for non aws resources (and this just happens to be the only attribute for it) would be good.

File: docs/users-manual/file-transfer.rst [hunk 4]
```diff
@@ -718,43 +718,53 @@ and to upload ``output.txt`` to ``my_private_files/output.txt`` on a second acco
...
+key (and nothing else), and one or more S3 URLs.  See the subsections below
+for specific examples.
+
+HTCondor does not currently support transferring entire buckets or directories
+from S3.  However, if you specify an ``s3://`` URL as the
+``output_destination``, that URL will be used a prefix for each output file's
+location; if you specify a URL ending a ``/``, it will be treated like a
+directory.
+
+You may also use S3 URLs in ``transfer_output_remaps``.
```
[the remark this answers is not in the document]
Comment (ChristinaLK) [reply docs/users-manual/file-transfer.rst:744]: This just seems like a non sequitur - "You may also" implies that you're continuing a thought, but I don't know what thought you're continuing.

File: docs/users-manual/file-transfer.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply docs/users-manual/file-transfer.rst]: I moved the lines about specifying a region out of the example and added a note that the name is misleading.

File: docs/users-manual/file-transfer.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply docs/users-manual/file-transfer.rst]: Suggestion incorporated.  Thanks.

File: docs/users-manual/file-transfer.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply docs/users-manual/file-transfer.rst]: Hopefully resolved by new draft.

**Review by ChristinaLK — APPROVED** [review]

**Review by timtheisen — APPROVED** [review]

**Pull request merged**